packages/svelte/src/internal/client/dom/elements/transitions.js

```diff
@@ -137,685 +64,252 @@ function css_to_keyframe(css) {
-		if (this.#delay === 0) {
-			this.#reverse();
-		} else {
-			this.#delay_current = this.#delay;
-			this.#delayed_reverse = true;
-		}
-	}
+	/** @type {DOMRect} */
+	let to;
 
-	cancel() {
-		active_tick_animations.delete(this);
-		const current = this.#current / this.#duration;
-		if (current > 0 && current < 1) {
-			const t = this.#reversed ? 1 : 0;
-			this.#tick_fn(t, 1 - t);
-		}
-	}
+	/** @type {import('#client').Animation | undefined} */
+	let animation;
 
-	finish() {
-		active_tick_animations.delete(this);
-		if (this.onfinish) {
-			this.onfinish();
-		}
-	}
+	/** @type {import('#client').AnimationManager} */
+	const manager = {
+		measure() {
+			from = element.getBoundingClientRect();
+		},
+		apply() {
+			to = element.getBoundingClientRect();
 
-	/** @param {number} time */
-	_update(time) {
-		let diff = time - this.#previous;
-		this.#previous = time;
-		if (this.#delay_current !== 0) {
-			const is_delayed = this.#delay_current === DELAY_NEXT_TICK;
-			let cancel = !this.#delayed_reverse;
-			this.#delay_current -= diff;
-			if (this.#delay_current < 0 || is_delayed || (this.#delay_current === 0 && this.#reversed)) {
-				const delay_diff = is_delayed ? 0 : -this.#delay_current;
-				this.#delay_current = 0;
-
-				if (this.#delayed_reverse) {
-					this.#delayed_reverse = false;
-					this.#reverse();
-				} else if (delay_diff !== 0 || this.#reversed) {
-					diff = delay_diff;
-				}
-				cancel = false;
-			} else if (this.#delay_current === 0) {
-				this.#delay_current = DELAY_NEXT_TICK;
-			}
-			if (cancel) {
-				return;
-			}
-		}
-		this.#current += this.#reversed ? -diff : diff;
-		let t = this.#current / this.#duration;
+			const options = get_fn()(element, { from, to }, get_params?.(), {}); // TODO what is the last argument?
 
-		if (t < 0) {
-			t = 0;
-		} else if (t > 1) {
-			t = 1;
-		}
+			animation?.abort();
 
-		if ((this.#reversed && t <= 0) || (!this.#reversed && t >= 1)) {
-			t = this.#reversed ? 0 : 1;
-			if (this.#delay_current === 0) {
-				active_tick_animations.delete(this);
-				if (this.onfinish) {
-					this.paused = true;
-					this.onfinish();
-				}
-			}
+			// TODO bail if `from` and `to` match
+			animation = animate(element, options, undefined, 1, () => {
+				animation = undefined;
+			});
 		}
-		this.#tick_fn(t, 1 - t);
-	}
-}
+	};
 
-/** @param {number} time */
-function handle_raf(time) {
-	for (const animation of active_tick_animations) {
-		if (!animation.paused) {
-			animation._update(time);
-		}
-	}
-	if (active_tick_animations.size !== 0) {
-		active_tick_ref = raf.tick(handle_raf);
-	} else {
-		active_tick_ref = undefined;
-	}
+	/** @type {import('#client').EachItemBlock} */ (current_each_item_block).a = manager;
 }
 
 /**
- * @param {{(t: number): number;(t: number): number;(arg0: number): any;}} easing_fn
- * @param {((t: number, u: number) => string)} css_fn
- * @param {number} duration
- * @param {string} direction
- * @param {boolean} reverse
+ * @template P
```

Comment:
needs some explanation about the nomenclature (intro outro etc) and how it's roughly working together